content/blog/automation-api/index.md

```diff
@@ -0,0 +1,521 @@
+    var num string
+    var animal string
+    handler := func(w http.ResponseWriter, r *http.Request) {
+        // Notice how we have the URL of randomNumberService
+        // available to consume here!
+        numResp := http.Get(randomNumberService.URL())
+        num = readResponse(numResp)
+
+        // Notice how we have the URL of randomAnimalService
+        // available to consume here!
+        animalResp, err := http.Get(randomAnimalService.URL())
+        animal = readResponse(animalResp)
+
+        fmt.Fprintf(w, "Wow, you got %s %ss!", num, animal)
+    }
+    r.HandleFunc("/", handler)
+    return r
+})
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+![Halloumi demo](halloumi.gif)
+
+Check out the full `halloumi` source in [TypeScript](https://github.com/pulumi/halloumi/tree/main/nodejs) and [Go](https://github.com/pulumi/halloumi/tree/main/go).
+
+Like any other framework, you can even set a breakpoint to debug orchestration logic or the pulumi program itself, including `Apply` callbacks.
+
+![Debuggin demo](debugging.gif)
+
+## Supercharged Ops Tooling
+
+Cloud infrastructure is beyond the days of driving deployments and infrastructure by hand, even though many of our tools are CLIs for manual workflows. Cloud engineering demands automated tools to keep up with growth, not just in terms of pace but also in complexity.
+
+Have a look at our temporary development VM provisioner for Azure `vmgr`. A simple Cobra CLI command creates a new stack with a VM ready to go:
+
+```go
+func NewAddCmd() *cobra.Command {
+    return &cobra.Command{
+        Use:   "add",
+        Short: "add deploys an additional additional vm stack",
+        Run: func(cmd *cobra.Command, args []string) {
+                stackName := fmt.Sprintf("vmgr%d", rangeIn(10000000, 99999999))
+                // create a new stack for our VM with an Inline program.
+                // the program is nil for now, but we'll set it later after retrieving some dependent outputs.
+                s, err := auto.NewStackInlineSource(ctx, stackName, projectName, nil /* Program */)
+                fmt.Println("ensuring network is configured...")
+                // lookout network and resource group info
+                subnetID, rgName, err := EnsureNetwork(ctx, projectName)
+                // set our program for the deployment with the resulting network info
+                stack.SetProgram(GetDeployVMFunc(subnetID, rgName))
+                fmt.Println("deploying vm...")
+                // deploy our VM stack
+                stdoutStreamer := optup.ProgressStreams(os.Stdout)
+
+                res, err := s.Up(ctx, stdoutStreamer)
+                if err != nil {
+                    fmt.Printf("Failed to deploy vm stack: %v\n", err)
+                    os.Exit(1)
+                }
+                fmt.Printf("deployed server running at public IP %s\n", res.Outputs["ip"].Value)
+            },
+        }
```

Comment:
I think you are missing a `}` / `cobra.Command{}` is not closed

Reply:
Turns out it was a formatting issue! Thanks for catching this.